src/shared/components/github-connect/GithubConnect.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useCallback, useState, useEffect } from 'react';
+
+import { Modal, Button } from 'ui';
+
+import { useAuthProvider } from 'core/auth';
+import { useCookiesProvider } from 'core/cookies';
+
+import csx from './GithubConnect.scss';
+
+interface State {
```

Comment:
Please change this to simple flag and state to status and setStatus - object is not needed here